pkg/webhook/clusterpropagationpolicy/validating.go

```diff
@@ -60,6 +60,12 @@ func (v *ValidatingAdmission) Handle(_ context.Context, req admission.Request) a
 			klog.Error(err)
 			return admission.Denied(err.Error())
 		}
+
+		if policy.Labels[policyv1alpha1.ClusterPropagationPolicyPermanentIDLabel] !=
+			oldPolicy.Labels[policyv1alpha1.ClusterPropagationPolicyPermanentIDLabel] {
+			return admission.Denied(fmt.Sprintf("the label %q is protected, any updates are prohibited",
+				policyv1alpha1.ClusterPropagationPolicyPermanentIDLabel))
```

Comment:
```suggestion
			return admission.Denied(fmt.Sprintf("label %s is immutable, it can only be set by the system during creation",
				policyv1alpha1.ClusterPropagationPolicyPermanentIDLabel))
```

Reply:
I tend not to use `%q` as it is used for printing integers. Feels tricky just for quotes.

Reply:
In addition, we still need to ensure the label's existence, even in the case of creation.

Reply:
I see the suggestion of @RainbowMango, I feel like it's ok, what do you think? @XiShanYongYe-Chang

Reply:
ok

Reply:
> In addition, we still need to ensure the label's existence, even in the case of creation.

sorry, I don't fully understand it. do you mean that we still need to validate it when `Request.Operation` is `CREATE`?

Reply:
I think maybe we don't need it.

Reply:
We have to ensure the label `clusterpropagationpolicy.karmada.io/permanent-id` exists, that's why I ask to introduce validation, we can not skip it even though it is already added by mutating. 
Note that, validation runs after mutation.

Reply:
The only scenario I can think of is that the user privately accesses the policy's mutating webhook, which may result in the policy being not labeled.

Reply:
In general, the `validation admission webhook` is used to guarantee all fields are set as expected, it usually doesn't assume the `mutating admission webhook` is already well processed.

Reply:
make sense, thanks

Reply:
so, how about the error message `label %s is required, it should not happen` for the case of creation?

Reply:
`label %s is required, it should be set by the mutating admission webhook during creation`.
How about this, we can give some useful hints for people to figure out the root cause.

Reply:
> I think maybe we don't need it.

@XiShanYongYe-Chang Why do you think we don't need it?

Reply:
I think it is good.